pkg/codegen/go/gen.go

```diff
@@ -1295,6 +1293,9 @@ func (pkg *pkgContext) getDefaultValue(dv *schema.DefaultValue, t schema.Type) (
 
 func (pkg *pkgContext) genResource(w io.Writer, r *schema.Resource, generateResourceContainerTypes bool) error {
 	name := resourceName(r)
```

Comment:
This reads weird and will confuse me when working with this code later. Does the place that calls `resourceName` func in `genResourceModule` also needs the renamed treatment? The renamed treatment could also be packaged into some func accepting pkg and `*schema.Resource`.

I'm noticing `pkg.renamed` as present before, but not used here.. Rename resourceName to `resourceNameBeforeNameDisambiguation`?

Reply:
Good catch, refactored and updated the tests.